addons/ofxEmscripten/src/ofxAppEmscriptenWindow.cpp

```diff
@@ -14,181 +14,79 @@ using namespace std;
 
 ofxAppEmscriptenWindow * ofxAppEmscriptenWindow::instance = NULL;
 
-// from http://cantuna.googlecode.com/svn-history/r16/trunk/src/screen.cpp
-#define CASE_STR(x,y) case x: str = y; break
-
-static const char* eglErrorString(EGLint err) {
-    string str;
-    switch (err) {
-        CASE_STR(EGL_SUCCESS, "no error");
-        CASE_STR(EGL_NOT_INITIALIZED, "EGL not, or could not be, initialized");
-        CASE_STR(EGL_BAD_ACCESS, "access violation");
-        CASE_STR(EGL_BAD_ALLOC, "could not allocate resources");
-        CASE_STR(EGL_BAD_ATTRIBUTE, "invalid attribute");
-        CASE_STR(EGL_BAD_CONTEXT, "invalid context specified");
-        CASE_STR(EGL_BAD_CONFIG, "invald frame buffer configuration specified");
-        CASE_STR(EGL_BAD_CURRENT_SURFACE, "current window, pbuffer or pixmap surface is no longer valid");
-        CASE_STR(EGL_BAD_DISPLAY, "invalid display specified");
-        CASE_STR(EGL_BAD_SURFACE, "invalid surface specified");
-        CASE_STR(EGL_BAD_MATCH, "bad argument match");
-        CASE_STR(EGL_BAD_PARAMETER, "invalid paramater");
-        CASE_STR(EGL_BAD_NATIVE_PIXMAP, "invalid NativePixmap");
-        CASE_STR(EGL_BAD_NATIVE_WINDOW, "invalid NativeWindow");
-        CASE_STR(EGL_CONTEXT_LOST, "APM event caused context loss");
-        default: str = "unknown error " + ofToString(err); break;
-    }
-    return str.c_str();
-}
-
-ofxAppEmscriptenWindow::ofxAppEmscriptenWindow()
-:display(NULL)
-,context(NULL)
-,surface(NULL)
-,bEnableSetupScreen(true){
-	instance = this;
 
+ofxAppEmscriptenWindow::ofxAppEmscriptenWindow(){
+	instance = this;
 }
 
 ofxAppEmscriptenWindow::~ofxAppEmscriptenWindow() {
-	// TODO Auto-generated destructor stub
+	if(context != 0){
+		emscripten_webgl_destroy_context(context);	
+	}
 }
 
 void ofxAppEmscriptenWindow::setup(const ofGLESWindowSettings & settings){
-	EGLint numConfigs;
-	EGLint majorVersion;
-	EGLint minorVersion;
-	EGLConfig config;
-	EGLint contextAttribs[] = { EGL_CONTEXT_CLIENT_VERSION, 2, EGL_NONE, EGL_NONE };
-	std::vector <EGLint> attribList =
-	   {
-		   EGL_RED_SIZE, EGL_DONT_CARE,
-		   EGL_GREEN_SIZE, EGL_DONT_CARE,
-		   EGL_BLUE_SIZE, EGL_DONT_CARE,
-		   EGL_ALPHA_SIZE, EGL_DONT_CARE,
-		   EGL_DEPTH_SIZE, EGL_DONT_CARE,
-		   EGL_STENCIL_SIZE, EGL_DONT_CARE,
-		   EGL_SAMPLE_BUFFERS, EGL_DONT_CARE,
-		   EGL_NONE
-	   };
+    setWindowShape(settings.getWidth(),settings.getHeight());
+    EmscriptenWebGLContextAttributes attrs;
+    emscripten_webgl_init_context_attributes(&attrs);
+
+/// when setting  explicitSwapControl to 0 it is emscripten that is in charge of swapping on each render call.
+    attrs.explicitSwapControl = 0;
+    attrs.depth = 1;
+    attrs.stencil = 1;
+    attrs.antialias = 1;
+    attrs.majorVersion = 2;
+    attrs.minorVersion = 0;
+    attrs.alpha = 0;
```

Comment:
should we be passing in any other settings from the settings argument?

Reply:
not sure, if the other settings are needed in some cases?